File: src/domain/models/Employee.ts
import { ColumnType, Generated, Insertable, Selectable, Updateable } from 'kysely';

export const EMPLOYEE_TABLE = 'employee';

export interface EmployeeTable {
  id: Generated<number>
  active: number;
  firstName: string;
  lastName: string;
  hireDate: ColumnType<Date | null, string | null, string | null>;
  phone: string | null;
  address: string | null;
  avatarUrl: string | null;
  departmentId: number | null
  
  // ColumnType<SelectType, InsertType, UpdateType>
  creationDate: ColumnType<Date, string, never>;
  modificationDate: ColumnType<Date, string, string>;
}

export type Employee = Selectable<EmployeeTable>
export type NewEmployee = Insertable<EmployeeTable>
export type UpdatedEmployee = Updateable<EmployeeTable>

File: src/wsample.domain/Models/Employee.cs
namespace wsample.domain.Models
{
    public class Employee
    {
        public int Id { get; set; }
        public int Active { get; set; }
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public DateTime? HireDate { get; set; }
        public string? Phone { get; set; }
        public string? Address { get; set; }
        public string? AvatarUrl { get; set; }
        public int? DepartmentId { get; set; }
        public DateTime CreationDate { get; set; }
        public DateTime ModificationDate { get; set; }
    }
}
